src/Sylius/Bundle/ApiBundle/Controller/UploadTaxonImageAction.php

```diff
@@ -0,0 +1,35 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Sylius Sp. z o.o.
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Bundle\ApiBundle\Controller;
+
+use Sylius\Bundle\ApiBundle\Creator\TaxonImageCreatorInterface;
+use Sylius\Component\Core\Model\TaxonImageInterface;
+use Symfony\Component\HttpFoundation\Request;
+
+/** @experimental */
+final class UploadTaxonImageAction
+{
+    public function __construct(private TaxonImageCreatorInterface $taxonImageCreator)
+    {
+    }
+
+    public function __invoke(Request $request): TaxonImageInterface
+    {
+        return $this->taxonImageCreator->create(
+            $request->attributes->get('code', ''),
+            $request->files->get('file'),
+            $request->request->get('type'),
+        );
+    }
```

Comment:
Do you plan to reuse the `TaxonImageCreator`? From what I see, this class lives in the `ApiBundle`, but it seems to not be coupled with any other class from the bundle. 

In the current shape, I see no difference between putting the logic from `TaxonImageCreator` directly here and keeping it in a separate class.

Reply:
Personally, I think that if it is possible, then no logic should be put into the controllers, they should only be used to call other services, which is what I have done here. Even if I wouldn't use this creator again, I still think this solution is a bit clearer and a bit easier to customize.